pilot/pkg/model/config.go

```diff
@@ -1062,6 +1088,81 @@ func (store *istioConfigStore) EndUserAuthenticationPolicySpecByDestination(inst
 	return out
 }
 
+// Set jwks_uri through openID discovery if it's not set in auth policy.
+func (store *istioConfigStore) SetAuthenticationPolicyJwksURIs(policy *authn.Policy) {
+	if policy == nil {
+		return
+	}
+
+	for _, method := range policy.Peers {
+		switch method.GetParams().(type) {
+		case *authn.PeerAuthenticationMethod_Jwt:
+			policyJwt := method.GetJwt()
+			uri, err := store.getJwksURI(policyJwt)
+			if err != nil {
+				log.Warnf("Failed to get jwks_uri for issuer %q: %v", policyJwt.Issuer, err)
+				continue
+			}
+			policyJwt.JwksUri = uri
+		}
+	}
+	for _, method := range policy.Origins {
+		policyJwt := method.GetJwt()
```

Comment:
Check/ASSERT method type is "jwt" before calling method.GetJwt()?

Reply:
From https://github.com/istio/api/blob/master/authentication/v1alpha1/policy.proto#L120, for now OriginAuthenticationMethod only has jwt field (while PeerAuthenticationMethod could be either jwt or mtls)

Reply:
Maybe add comments saying that "JWT is only allowed authentication method type for Origin"?

Reply:
added.